LSLib/LS/FileManager.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.IO;
+
+namespace LSLib.LS
+{
+    internal class FileManager
+    {
+        public static bool TryToCreateDirectory(string path)
+        {
+            if (path.Length <= 0)
+            {
+                return false;
+            }
+
+            try
+            {
+                // verify this is a path
+                string outputPath = Path.GetFullPath(path);
+
+                // get only path to directory if a filename was found
+                if (Path.GetExtension(outputPath) != string.Empty)
```

Comment:
I could create a directory named "C:\Stuff.txt" and it would fail this check, since it looks like a filename but is a directory. (`C:\Windows\Microsoft.NET` is a real-life example of this). I think `TryToCreateDirectory` should know if the parameter passed is a filename (ie. if it should always strip the last part of the path), or if it is a directory (so it should keep the last part), as speculating about the type of the path can easily lead to errors.

Reply:
Fixed but not committed yet.

Also added checks to ensure the path is absolute and within MAX_PATH - 12.